### ISSUE TITLE
 Illegal instruction: 4 when importing tensorflow

### ISSUE THREAD

**Issue by vrl2** [tensorflow/tensorflow#25822]
I've installed tensorflow (Version: 1.12.0) using pip3 on MacOS High Sierra 10.13.6, and it crashes on import with Illegal Instruction: 4. I get the same error using any version of python 3.7 and 3.6.8. Same error even after downgrading to tensorflow 1.5 and numpy 1.13. Identical issue whether I install through homebrew or pip3. Would appreciate any help to get it working.
Thanks.
```
$ python3
Python 3.6.8 (v3.6.8:3c6b436a57, Dec 24 2018, 02:04:31) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow
Illegal instruction: 4
```
 I'm also attaching the full error log, with all the details of my system configuration. 
[ill4.txt](https://github.com/tensorflow/tensorflow/files/2873724/ill4.txt)

**Comment by ymodak** [comment]
What is your current numpy version?
The numpy version should be  >= 1.14.5 and < 2.0

**Comment by vrl2** [comment]
Numpy version is 1.16.1. Also tried with version 1.14.5 with the same
result.

**Comment by gunan** [comment]
what is your CPU make/model?
TF requires CPUs that have AVX instruction sets. If you are running on an old machine, it will fail to load.

**Comment by vrl2** [comment]
The CPU is an Intel Xeon X5680, which does not have AVX instruction sets.
Is it possible to run tensorflow in a conda virtual environment with this
processor?

**Comment by gunan** [comment]
Not the prepackaged versions.
If you build TF from sources, you can run TF on this machine:
https://www.tensorflow.org/install/source

**Comment by TomHeaven** [comment]
I'm facing the same problem using Tensorflow 1.12.0 built from source. The problem persists in Python 3.7 but does not exist in Python2.7. So maybe it's bug related to Python 3.7.

**Comment by vrl2** [comment]
As a workaround, I able to run tensorflow, installed through anaconda, in
an anaconda virtual environment (both python 3.7 and 2.7):
https://www.anaconda.com/tensorflow-in-anaconda/

**Comment by ymodak** [comment]
@TomHeaven TF 1.12 does not support Python 3.7 however TF 1.13.0-rc2 supports Python 3.7. You may want to give it a try.

**Comment by ikrunalpatel** [comment]
@vrl2 Try using Tensorflow 1.5 with Python 2.7. I got the same error and I have tried Tensorflow versions 1.5, 1.5.1, 1.7, and 1.13 but only v1.5 worked. You can try doing that.
My system info: MacOS Mojave 10.14; Python 2.7; TensorFlow 1.5

**Comment by ymodak** [comment]
Closing this issue since its resolved. Thanks!

**Comment by wqw547243068** [comment]
> 

Both TF 1.13.2 and TF 2.10.0 Still doesn't support python 3.7.9 in Macos (12.4 (21F79))

```shell
Python 3.7.9 (v3.7.9:13c94747c7, Aug 15 2020, 01:31:08) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow 
<font color='red'>Illegal instruction: 4</font>
```

<img width="247" alt="image" src="https://user-images.githubusercontent.com/3257257/191433539-b5bfb887-cc12-4c11-ab42-def1b6046e17.png">

**Comment by ChemGuy88** [comment]
It seems like Tensorflow has limited support on MacOS. [Anaconda says](https://www.anaconda.com/blog/apple-silicon-transition) that TensorFlow is available for Python 3.8 on MacOS, but that hasn't worked for me. Below are my computer details.

```
Software:

    System Software Overview:

      System Version: macOS 13.6 (22G120)
      Kernel Version: Darwin 22.6.0
      Boot Volume: Macintosh HD
      Boot Mode: Normal
      Computer Name: FE15-B200-M10
      User Name: Autore,Herman F (herman)
      Secure Virtual Memory: Enabled
      System Integrity Protection: Enabled
      Time since boot: 1 day, 48 minutes

Hardware:

    Hardware Overview:

      Model Name: MacBook Pro
      Model Identifier: MacBookPro18,1
      Model Number: MK183LL/A
      Chip: Apple M1 Pro
      Total Number of Cores: 10 (8 performance and 2 efficiency)
      Memory: 16 GB
      System Firmware Version: 10151.1.1
      OS Loader Version: 8422.141.2
      Serial Number (system): KC6YQ3YL43
      Hardware UUID: A940A668-5DCA-540B-9A89-BD59EF1F8968
      Provisioning UDID: 00006000-000268342123801E
      Activation Lock Status: Disabled
```

I've also tried older versions of Python and tensorflow, following [this post's](https://stackoverflow.com/questions/49084934/illegal-instruction-4-when-importing-tensorflow-in-python-3-6) advice, but with no success.

The TensorFlow forums also have little to offer in the way of [solutions](https://discuss.tensorflow.org/t/illegal-instruction-importing-tensorflow/13965).
